cabal-install/src/Distribution/Client/IndexUtils.hs

```diff
@@ -726,8 +778,12 @@ withIndexEntries verbosity (RepoIndex _repoCtxt (RepoLocalNoIndex (LocalRepo nam
                     Nothing -> die' verbosity $ "Cannot read .cabal file inside " ++ file
 
     info verbosity $ "Entries in file+noindex repository " ++ unRepoName name
-    for_ entries $ \(CacheGPD gpd _) ->
-        info verbosity $ "- " ++ prettyShow (package $ Distribution.PackageDescription.packageDescription gpd)
+    for_ entries $ \case
+        NoIndexCachePreference deps ->
+            for_ deps $ \dep ->
+              info verbosity ("* " ++ prettyShow dep)
```

Comment:
I think we need clearer separation.

I'd rather go over entries twice (or `partitionEithers` or ...) and print preferred versions independently under their own header (i.e. `info verbosity $ "Prefferred versions in ...`).

Reply:
Output is now better grouped